src/hardware/input/mouseif_dos_driver_state.h

```diff
@@ -0,0 +1,393 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2025-2025  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_MOUSEIF_DOS_DRIVER_STATE_H
+#define DOSBOX_MOUSEIF_DOS_DRIVER_STATE_H
+
+#include "dos_inc.h"
+#include "dos_windows.h"
+#include "mouse_common.h"
+
+static constexpr auto MaxMouseButtons  = 3;
+static constexpr auto MaxUpdateRegions = 2;
+
+static constexpr uint8_t CursorSize = 16;
+
+enum class MouseCursor : uint8_t { Software = 0, Hardware = 1, Text = 2 };
+
+class MouseDriverState final : public MemStruct {
+public:
+	explicit MouseDriverState(const uint16_t segment)
+	{
+		SetPt(segment);
+	}
+
+	// The state access object shouldn't be copied around
+	MouseDriverState(const MouseDriverState& other)            = delete;
+	MouseDriverState& operator=(const MouseDriverState& other) = delete;
+
+	static uint16_t GetSize()
+	{
+		static_assert(sizeof(State) <= UINT16_MAX);
+		return static_cast<uint16_t>(sizeof(State));
+	}
+
+	std::vector<uint8_t> ReadBinaryData() const;
+	void WriteBinaryData(const std::vector<uint8_t>& source);
+
+	uint16_t GetWin386StateOffset() const;
+	uint16_t GetWin386StartupOffset() const;
+	uint16_t GetWin386InstancesOffset() const;
```

Comment:
Looking at the names below, shouldn't these be `Win386State_GetOffset()`, etc? Just wondering.